src/core/loop_induction_variables/src/InductionVariables.cpp

```diff
@@ -89,27 +89,78 @@ InductionVariableManager::InductionVariableManager(LoopTree *loopNode,
        * First, let's check if the PHI node can be analyzed by the SCEV
        * analysis.
        */
+      auto sccContainingIV = sccdag.sccOfValue(&phi);
       auto noelleDeterminedValidIV = true;
+      auto stepMultiplier = 0;
+      InductionVariable *IV = nullptr;
       if (!SE.isSCEVable(phi.getType())) {
         noelleDeterminedValidIV = false;
-
       } else {
 
         /*
          * Fetch the SCEV and check if it suggests this is an induction
          * variable.
          */
         auto scev = SE.getSCEV(&phi);
-        if (!scev || scev->getSCEVType() != SCEVTypes::scAddRecExpr) {
+        if (!scev) {
+          noelleDeterminedValidIV = false;
+        } else if (scev->getSCEVType() != SCEVTypes::scAddRecExpr) {
           noelleDeterminedValidIV = false;
+
+          PHINode *internalPHI;
+          auto found = false;
+          for (auto internalNode = sccContainingIV->begin_internal_node_map();
```

Comment:
Can there be cases that follow the pattern this code region aims to catch that have no IVs?
In other words, can there be false positive in your algorithm?